include/opencensus.hrl

```diff
@@ -103,3 +103,40 @@
           message :: unicode:unicode_binary()
          }).
 
+
+%% stats types
+
+-type measure_name() :: atom().
+
+-type measure_unit() :: binary() | string().
+
+-type measure_description() :: binary() | string().
+
+-record(measure, {
+          name        :: measure_name(),
+          unit        :: measure_unit(),
+          description :: measure_description()
+         }).
+
+-type measure() :: #measure{}.
+
+-type view_name() :: atom() | binary() | string().
+-type view_description() :: binary() | string().
+
+-type tag_key() :: atom().
```

Comment:
Atom might make more sense, I had gone with:

```erlang
-type key() :: unicode:latin1_charlist().
-type value() :: unicode:latin1_charlist().
-type tags() :: #{key() => value()}.
```

based on the spec that requires they be ascii, then the checks to verify each is just:

```erlang
verify_key(Key) ->
    KeyLength = erlang:length(Key),
    KeyLength > 0 andalso KeyLength < 256 andalso
        io_lib:printable_latin1_list(Key).

verify_value(Value) ->
    erlang:length(Value) < 256 andalso io_lib:printable_latin1_list(Value).
```

atom would save space but mean we'd only want to do verification at the end and not be able to block adding tags that don't fit the requirements.

Reply:
depending on how exactly 'adding' is done we can check atom's name. Saving space wasn't my main motivation here. I went with atoms mostly because of the lookup/hash speed (incl in ets).

Reply:
Ah, it uses ets lookups on the tag key? Hm, maybe should stick with atoms then.

Reply:
yes, tags used in the ets keys.